ir/src/boundary_constraints.rs

```diff
@@ -33,13 +33,21 @@ impl BoundaryConstraints {
     }
 
     /// Returns all of the boundary constraints for the first row of the main trace.
-    pub fn main_first(&self) -> Vec<&BoundaryExpr> {
-        self.main_first.values().collect()
+    pub fn main_first(&self) -> Vec<(usize, &BoundaryExpr)> {
+        let mut constraints = Vec::new();
+        for (k, v) in &self.main_first {
+            constraints.push((*k, v))
+        }
+        constraints
```

Comment:
nit: I would replace this with the following code to simplify it, and I would make this change for all 4 methods modified in this file.

```
        self.main_first.iter().map(|(k, v)| (*k, v)).collect()
```

Reply:
Thanks. Fixed.